### PR TITLE
mtd-utils: update build.sh to a new convention

### PR DESCRIPTION
**Pull request by damianloew**
## Motivation and Context

fix building zynq/ull targets on macOS after changes in phoenix-rtos-build/build.sh

JIRA: CI-340

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## How Has This Been Tested?

- [ ] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [ ] Tested by hand on: (list targets here).

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing linter checks and tests passed.
- [ ] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [ ] This PR needs additional PRs to work ().
- [ ] I will merge this PR by myself when appropriate.

### CODE REVIEW

File: mtd-utils/build.sh [hunk 1]
```diff
@@ -11,7 +11,7 @@ if [ "$(uname)" = "Darwin" ]; then
 	mkdir -p "$PREFIX_MTD_UTILS_BUILD"
 
 	pushd "$PREFIX_MTD_UTILS_BUILD"
-	if [ -n "$CLEAN" ]; then
+	if [ "$B_CLEAN" = "y" ]; then
```
Comment (nalajcie) [inline mtd-utils/build.sh:14]: this should not depend on `B_CLEAN` variable but on existence of target files

**Review by nalajcie — CHANGES_REQUESTED** [review]
Thanks for noticing this.

As this is a host tool I would gladly move this step to host preparation in docs and remove mtd-utils from `phoenix-rtos-project` altogether.

From what I see - getting `mkfs.jffs2` on Darwin is problematic so we can leave it as-is for now, but IMHO we should at least move it to hostsutils repo in the near future.

File: mtd-utils/build.sh (hunk 1 above)
Comment (damianloew) [reply mtd-utils/build.sh]: Seems to be a better solution - updated. I'd merge it as it is for now.

File: mtd-utils/build.sh [hunk 2]
```diff
@@ -7,11 +7,12 @@ if [ "$(uname)" = "Darwin" ]; then
 
 	PREFIX_MTD_UTILS="${PREFIX_PROJECT}/mtd-utils"
 	PREFIX_MTD_UTILS_BUILD="${PREFIX_PROJECT}/_build/host-generic-pc/mtd-utils"
```
Comment (nalajcie) [inline mtd-utils/build.sh:9]: maybe use `PREFIX_BUILD_HOST` (here and everywhere below also)?

Comment (damianloew) [reply mtd-utils/build.sh:9]: Good idea :+1:

Comment (nalajcie) [reply mtd-utils/build.sh:9]: `PREFIX_BUILD_HOST` is being exported from `build.sh`, don't define it by yourself. 

Also - missed the line above.

Comment (damianloew) [reply mtd-utils/build.sh]: Thanks, done.

**Review by nalajcie — APPROVED** [review]
Ok as a stop-gap for now.

**Pull request merged**